### PR TITLE
Adds more ACME solver docs

### PR DESCRIPTION
**Pull request by JoshVanL**
Signed-off-by: JoshVanL <vleeuwenjoshua@gmail.com>
/assign @munnerz

### CODE REVIEW

File: content/en/docs/configuration/acme/_index.md [hunk 1]
```diff
@@ -80,15 +80,25 @@ this solver types, visit their respective documentation -
...
+There are three selector types that can be used to form the requirements that a
+`Certificate` must meet in order to be selected for a solver - `matchLabels`,
+`dnsNames` and `dnsZones`. You can have any number of these thee selectors on a
+single selector.
+
+
+#### Match Labels
+
+The `matchLabel` selector requires that all `Certificates` match at least one of
+the labels that are defined in the string map list of that stanza. For example,
```
Comment (munnerz) [inline content/en/docs/configuration/acme/_index.md]: `match all of the labels` (they are ANDed)

File: content/en/docs/configuration/acme/_index.md [hunk 2]
```diff
@@ -110,21 +117,92 @@ spec:
             key: apikey
       selector:
         matchLabels:
-          use-cloudflare-solver: "true"
+        - "use-cloudflare-solver": "true"
+        - "email": "user@example.com"
```
Comment (munnerz) [inline content/en/docs/configuration/acme/_index.md]: This is not a list, it is a `map[string]string`, so should be:

```
matchLabels:
  key: value
```

File: content/en/docs/configuration/acme/_index.md [hunk 3]
```diff
@@ -110,21 +117,92 @@ spec:
...
+```
+
+#### DNS Names
+
+The `dnsNames` selector is used to define the exact DNS challenges so solve.
```
Comment (munnerz) [inline content/en/docs/configuration/acme/_index.md]: `a list of exact DNS names that should be mapped to a solver` maybe?

File: content/en/docs/configuration/acme/_index.md [hunk 4]
```diff
@@ -110,21 +117,92 @@ spec:
...
+This means that `Certificates` containing any of these DNS names will be
+selected.  If a match is found, a `dnsNames` selector will take precedence over
+a [`dnsZones`](./#dns-zones) selector. If multiple solvers match with the same
+`dnsNames` value, the solver with the most matching labels in
+[`matchLabels`](./#match-labels) will be selected. If neither has more matches,
```
Comment (munnerz) [inline content/en/docs/configuration/acme/_index.md]: Can we omit the `./` prefix here and just link direct to `#match-labels` as this is in the same doc?

File: content/en/docs/configuration/acme/_index.md [hunk 5]
```diff
@@ -110,21 +117,92 @@ spec:
...
+the solver defined earlier in the list will be selected.
+
+The following example will solve challenges of `Certificates` with DNS names
+`exmaple.com` and `*.exmaple.com` for these domains.
```
Comment (munnerz) [inline content/en/docs/configuration/acme/_index.md]: `example.com` (both times)

File: content/en/docs/configuration/acme/_index.md [hunk 6]
```diff
@@ -110,21 +117,92 @@ spec:
...
+
+> Note: `dnsNames` take an exact match and do not resolve wildcards, meaning the
+> following issuer *will not* solve for DNS names such as `foo.example.com`.
```
Comment (munnerz) [inline content/en/docs/configuration/acme/_index.md]: Add:

```
Use the `dnsZones` selector type to match all subdomains within a zone.
```

File: content/en/docs/configuration/acme/_index.md [hunk 7]
```diff
@@ -110,21 +117,92 @@ spec:
...
+        - '*.example.com'
 ```
 
-In order to utilize the configured CloudFlare DNS01 solver, you must add the
-``use-cloudflare-solver: "true"`` label to your `Certificate` resources.
+#### DNS Zones
 
-### Using multiple solvers for a single certificate
+The `dnsZones` stanza defines a list of DNS zones that can be solved by this
+solver. This means that DNS names defined here will solve for that domain,
```
Comment (munnerz) [inline content/en/docs/configuration/acme/_index.md]:
```
If a DNS name is an exact match, or a subdomain of any of the specified `dnsZones`, this solver will be used, unless a more specific [dnsNames](#dns-names) match is configured.
```

Maybe a bit clearer?

File: content/en/docs/configuration/acme/_index.md [hunk 8]
```diff
@@ -110,21 +117,92 @@ spec:
...
+*and*, for all sub domains. The most specific DNS zone match specified will take
+precedence over other `dnsZone` matches, meaning a solver specifying
```
Comment (munnerz) [inline content/en/docs/configuration/acme/_index.md]: `dnsZones`

**Comment by JoshVanL** [comment]
@munnerz Should be good now
/unassign
/assign @munnerz

**Pull request merged**